### PR TITLE
Add support for skipping sending usage pixels for remote messages

### PR DESCRIPTION
**Pull request by ayoy**
Task/Issue URL: https://app.asana.com/0/1201621708115095/1207841204698435/f

**Description**:
This change allows to skip sending usage pixels for a given remote message if that's stated in the RMF config.

**Steps to test this PR**:
See [BSK PR](https://github.com/duckduckgo/BrowserServicesKit/pull/902) for steps to test.

**Definition of Done (Internal Only)**:

* [ ] Does this PR satisfy our [Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f)?

**Copy Testing**:

* [ ] Use of correct apostrophes in new copy, ie `’` rather than `'`

**Orientation Testing**:

* [ ] Portrait
* [ ] Landscape

**Device Testing**:

* [ ] iPhone SE (1st Gen)
* [ ] iPhone 8
* [ ] iPhone X
* [ ] iPhone 14 Pro
* [ ] iPad

**OS Testing**:

* [ ] iOS 15
* [ ] iOS 16
* [ ] iOS 17

**Theme Testing**:

* [ ] Light theme
* [ ] Dark theme

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: DuckDuckGo/RemoteMessagingClient.swift [hunk 1]
```diff
@@ -34,7 +34,7 @@ final class RemoteMessagingClient: RemoteMessagingProcessing {
         static let minimumConfigurationRefreshInterval: TimeInterval = 60 * 60 * 4
         static let endpoint: URL = {
 #if DEBUG
-            URL(string: "https://raw.githubusercontent.com/duckduckgo/remote-messaging-config/main/samples/ios/sample1.json")!
+            URL(string: "https://www.jsonblob.com/api/1263462242434539520")!
```
Comment (quanganhdo) [inline DuckDuckGo/RemoteMessagingClient.swift]:
```suggestion
            URL(string: "https://raw.githubusercontent.com/duckduckgo/remote-messaging-config/main/samples/ios/sample1.json")!
```

File: DuckDuckGo/RemoteMessagingDebugViewController.swift [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+                Text("This list contains messages that have been shown plus at most 1 message that is scheduled for showing. There may be more messages in the config that will be presented, but they haven't been processed yet.")
+            }
+        }
+        .navigationTitle("\(model.messages.count) Remote Messages")
+        .toolbar {
+            if #available(iOS 15, *) {
+                Button("Delete All", role: .destructive) {
+                    model.deleteAll()
+                }
+            }
```
Comment (quanganhdo) [inline DuckDuckGo/RemoteMessagingDebugViewController.swift]:
```suggestion
                Button("Delete All", role: .destructive) {
                    model.deleteAll()
                }
```

Deployment target is 15 already 👏

**Review by quanganhdo — APPROVED** [review]
Both `isMetricsEnabled` and the new Debug screen work great.

File: DuckDuckGo/RemoteMessagingClient.swift (hunk 1 above)
Comment (ayoy) [reply DuckDuckGo/RemoteMessagingClient.swift]: Absolutely :)

File: DuckDuckGo/RemoteMessagingDebugViewController.swift (hunk 2 above)
Comment (ayoy) [reply DuckDuckGo/RemoteMessagingDebugViewController.swift]: Ah amazing :) This was a copy-paste from History debug menu so let me update it there too!

**Pull request merged**